### PR TITLE
FDTN-689: Updated node base image

### PR DESCRIPTION
**Pull request by Yatharth0045**

### CODE REVIEW

File: node/Dockerfile.node_builder12 [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ENV PATH $PATH:$GOBIN
+
+ARG GITHUB_TOKEN
+
+WORKDIR /root
+
+RUN npm install -g forever && \
+    apk add --no-cache --update python3 py-pip ca-certificates musl-dev git go make build-base && \
+    pip install --upgrade awscli && \
+    apk -v --purge del py-pip
```
Comment (RoryH) [inline node/Dockerfile.node_builder12:20]:
```suggestion
    pip install --upgrade awscli
```

This command `apk -v --purge del py-pip` was removed in my latest PR... It causes a python dependency `six` to be removed which the awscli depends on at runtime.

File: node/build-node_builder [hunk 2]
```diff
@@ -27,13 +28,19 @@ if node_tag == ""
...
+end
+
 if ENV[github_token_var_name] == nil
   # GITHUB_TOKEN is required to pull dependencies for building Flow `dev` tool
   puts "ERROR: Required environment variable '#{github_token_var_name}' is missing"
   exit(1)
 end
 
-node_image_tag = "flowdocker/node_builder:%s" % node_tag
+node_image_tag = "flowdocker/node_builder%s:%s" % [node_version, node_tag]
```
Comment (gheine) [inline node/build-node_builder:43]: would prefer naming this "flowdocker/node%s_builder:%s" % [node_version, node_tag]

Comment (gheine) [reply node/build-node_builder:43]: eg node12_builder instead of node_builder12

Comment (RoryH) [reply node/build-node_builder:43]: Yeah, I prefer @gheine 's suggestion also. 👍🏻

File: node/build.go [hunk 3]
```diff
@@ -8,8 +8,10 @@ func main() {
 	executor := executor.Create("docker-node")
 
 	executor = executor.Add("dev tag")
-	executor = executor.Add("./build-node `sem-info tag latest`")
-	executor = executor.Add("./build-node_builder `sem-info tag latest`")
+	executor = executor.Add("./build-node `sem-info tag latest` 12")
+	executor = executor.Add("./build-node_builder `sem-info tag latest` 12")
+	executor = executor.Add("./build-node `sem-info tag latest` 16")
+	executor = executor.Add("./build-node_builder `sem-info tag latest` 16")
```
[the remark this answers is not in the document]
Comment (tgamyatharth) [reply node/build.go:14]: Just thinking to give @benwaffle  a chance to look into it once. I'll talk to him async on this.

[the remark this answers is not in the document]
Comment (Yatharth0045) [reply node/build.go:14]: Updated @gheine

File: node/build-node_builder [hunk 4]
```diff
@@ -62,11 +69,10 @@ begin
   puts ""
   puts "Completed build of #{node_image_tag}"
 
-  if confirm("Enter 'Y' to push this image (%s) to Docker hub" % node_image_tag)
```
Comment (gheine) [inline node/build-node_builder:65]: remove the confirm function from helpers.rb if it's not being used any more

**Review by gheine — CHANGES_REQUESTED** [review]
See above

**Review by gheine — APPROVED** [review]
lgtm

**Pull request merged**